planner/core/logical_plan_builder.go

```diff
@@ -6693,7 +6714,8 @@ func (b *PlanBuilder) buildCte(ctx context.Context, cte *ast.CommonTableExpressi
 		}
 		b.buildingRecursivePartForCTE = saveCheck
 	} else {
-		p, err = b.buildResultSetNode(ctx, cte.Query.Query)
+		//When the query is CTE, we will turn the var to true.
```

Comment:
The comment is meaningless, you can remove it.

Reply:
OK